src/ValueObject/DateValueObject.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme\ValueObject;
+
+use Drupal\Component\Datetime\DateTimePlus;
+use Drupal\Core\Datetime\DrupalDateTime;
+
+/**
+ * Handle information about a date.
+ */
+class DateValueObject extends ValueObjectBase implements DateValueObjectInterface {
+
+  /**
+   * DateValueObject constructor.
+   *
+   * @param string $day
+   *   The date day.
+   * @param string $month
+   *   The date month.
+   * @param string $year
+   *   The date year.
+   * @param string $variant
+   *   The pattern variant.
+   */
+  private function __construct(string $day, string $month, string $year, string $variant = 'default') {
```

Comment:
I'd keep the current approach with setters and getters for now, we have a follow up issue to consider the alternative proposed by @drupol , until that's not discussed and agreed we should not change direction. Refactorying would be relatively quick anyway and can be assigned to a person different than @drupol so to spread more the knowledge about the new approach, if accepted.

Reply:
See #151

Reply:
@ademarco I'm using variables in the constructor, nothing else for now... I guess this comment is outdated then.
I removed the "withers" as well.